### DISCUSSION TITLE
How to testing  api routes

### DISCUSSION THREAD

**Discussion by lionelB** [sveltejs/kit#9936] [Q&A]
Hello, 

I have a bunch of api routes (ex: `/routes/api/endpoint/+server.ts`) and I cant find a way to test that.
Is there a recommanded / documented way to do that. I don't find litterature about that. 
Do you have some tips to share ?
thanks

ps : It would be awesome that testing is also present in documentation.

**Comment by stephane-vanraes** [comment]
To test API Routes you can use testing suite that allows to make fetch requests.  

Personally I just use [Postman](https://www.postman.com/) because it allows me to easily set headers, group the tests, inspect the results and run them together (and with a simple switch change targets from localhost to a remote server)

**Reply by lionelB** [reply]
thanks for the reply. I realised it was not clear, I was talking about automated tests (using vitest, jest or other test runner)

**Comment by teemingc** [comment]
Here's an example repo using [Playwright](https://playwright.dev) that tests an endpoint:
https://stackblitz.com/github/s3812497/sveltekit-playwright

You can run the tests with `npm run test` or `pnpm test`. The test is stored in `/tests/test.ts`.

More example tests available in this very repository! https://github.com/sveltejs/kit/blob/master/packages/kit/test/apps/basics/test/server.test.js
